cpp/ql/src/semmle/code/cpp/Location.qll

```diff
@@ -128,7 +128,7 @@ deprecated library class LocationExpr extends Location, @location_expr { }
  * Gets the length of the longest line in file `f`.
  */
 pragma[nomagic]
-private int maxCols(File f) { result = max(Location l | l.getFile() = f | l.getEndColumn()) }
+private int maxCols(File f) { result = max(Location l | l.getFile() = f | [l.getStartColumn(), l.getEndColumn()]) }
```

Comment:
```suggestion
private int maxCols(File f) { result = max(Location l | l.getFile() = f | l.getStartColumn().maximum(l.getEndColumn())) }
```

Untested (sorry). As I remember it, `maxCols` is a somewhat slow predicate. I don't know if my proposed change makes it faster or slower, but it might be worth trying.

Reply:
It's not super-quick, but I don't think it's problematic like `subsumes`.

I'll try your suggestion.

Reply:
Yep, your version seems to be a touch quicker.